packages/cbv/src/Base.tsx

```diff
@@ -0,0 +1,98 @@
+ * @typeparam TSelector - the type definition for the Selector.
+ */
+export interface CBVOptions<TAction, TSelector> {
+  actionCreator: ActionCreator<TAction> /** The action creator function to use */;
+  returnPropName: string /** The name of the prop used in mapStateToProps which represents the variable fetched from the redux state */;
+  dispatchPropName: string /** The name of the prop used in mapDispatchToProps which represents the dispatch-able action creator */;
+  selector: TSelector /** The selector function to use when getting the value from the Redux store */;
+}
+
+/**
+ * Base class
+ *
+ * This class is initialized with the following parameters:
+ *    - component: a react component
+ *    - options: an object representing options for Base
+ *
+ * The class's render method will then return the same component wrapped in a
+ * higher order component that has been connected to the redux store, and which
+ * knows how to fetch data for display.
+ *
+ * The end goal is a quick and convenient way to display a list of objects without
+ * having to worry about boilerplate code that deal with fetching objects and
+ * storing them in a Redux store.
+ *
+ * Every method in this class can and should be overwritten to cater to custom needs.
+ *
+ * @typeparam ActionType - the type definition for the Action.
+ * @typeparam SelectorType - the type definition for the Selector.
+ * @typeparam PropsType - the type definition for the props that the connected component takes.
+ * @typeparam RootState - the type definition for the Redux store (global state).
+ */
+export abstract class Base<ActionType, SelectorType, PropsType, RootState = Registry> {
+  public Component: React.ElementType;
+  public options: CBVOptions<ActionType, SelectorType>;
+
+  /** constructor */
+  constructor(component: React.ElementType, options: CBVOptions<ActionType, SelectorType>) {
+    this.Component = component;
+    this.options = options;
+  }
+
+  /**
+   * This function returns a Higher Order component whose job is to wrap around
+   * the target component, and pass on props to it.
+   *
+   * This must be implemented.
+   */
+  public abstract getHOC(): React.ComponentType<any>; // TODO: find a way to remove any here
+
+  /**
+   * The mapDispatchToProps function
+   * You may override this for more custom needs.
+   */
+  public getMapDispatchToProps() {
+    return { [this.options.dispatchPropName]: this.options.actionCreator };
+  }
+
+  /**
+   * The mapStateToProps function
+   *
+   * This must be implemented.
+   */
+  public abstract getMapStateToProps(): (
+    state: RootState,
+    ownProps: PropsType
+  ) => { [key: string]: any };
+
+  /**
+   * This function simply connects the Higher Order Component to the redux
+   * store.
+   */
+  public getConnectedHOC() {
+    /** map dispatch to props */
+    const mapDispatchToProps = this.getMapDispatchToProps();
+    /** map state to props */
+    const mapStateToProps = this.getMapStateToProps();
```

Comment:
Then this line depends on the child classes defining their own `getMapStateToProps`?

Reply:
The way I think about it, is that for you to get here you will have instantiated an object using this class.  And for you to be able to do that, you need to have implemented the Base class.  And doing that means you will definitely need to implement `getMapStateToProps` or the Typescript compiler will have complained.  Makes sense?

Reply:
> is that for you to get here you will have instantiated an object using this class.

Yup, that is pretty much what I thought :)